lib/index.js

```diff
@@ -25,804 +25,89 @@ exports.version = zmq.zmqVersion();
- *
- * @param {String} type
- * @param {Object} options
- * @return {Socket}
- * @api public
- */
-
-exports.socket =
-exports.createSocket = function(type, options) {
-  var sock = new Socket(type);
-  for (var key in options) sock[key] = options[key];
-  return sock;
-};
-
-exports.Context.setMaxThreads = function(value) {
-  if (!zmq.ZMQ_CAN_SET_CTX) {
-    throw new Error('Setting of context options disabled, check zmq version is >= 3.2.1 and recompile this addon');
-  }
-  var defaultCtx = defaultContext();
-  defaultCtx.setOpt(zmq.ZMQ_IO_THREADS, value);
-};
-
-exports.Context.getMaxThreads = function() {
-  if (!zmq.ZMQ_CAN_SET_CTX) {
-    throw new Error('Getting of context options disabled, check zmq version is >= 3.2.1 and recompile this addon');
-  }
-  var defaultCtx = defaultContext();
-  return defaultCtx.getOpt(zmq.ZMQ_IO_THREADS);
-};
-
-exports.Context.setMaxSockets = function(value) {
-  if (!zmq.ZMQ_CAN_SET_CTX) {
-    throw new Error('Setting of context options disabled, check zmq version is >= 3.2.1 and recompile this addon');
-  }
-  var defaultCtx = defaultContext();
-  defaultCtx.setOpt(zmq.ZMQ_MAX_SOCKETS, value);
-};
-
-exports.Context.getMaxSockets = function() {
-  if (!zmq.ZMQ_CAN_SET_CTX) {
-    throw new Error('Getting of context options disabled, check zmq version is >= 3.2.1 and recompile this addon');
-  }
-  var defaultCtx = defaultContext();
-  return defaultCtx.getOpt(zmq.ZMQ_MAX_SOCKETS);
-};
-
-/**
- * JS based on API characteristics of the native zmq_proxy()
- */
-
-function proxy (frontend, backend, capture){
-  switch(frontend.type+'/'+backend.type){
+function proxy(frontend, backend, capture) {
+  switch (frontend.type + '/' + backend.type) {
     case 'push/pull':
     case 'pull/push':
     case 'xpub/xsub':
-      if(capture){
-
-        frontend.on('message',function (msg){
+      if (capture) {
+        frontend.on('message',function (msg) {
           backend.send(msg);
         });
 
-        backend.on('message',function (msg){
+        backend.on('message', function (msg) {
           frontend.send(msg);
 
           //forwarding messages over capture socket
           capture.send(msg);
         });
-
       } else {
-
         //no capture socket provided, just forwarding msgs to respective sockets
-        frontend.on('message',function (msg){
+        frontend.on('message', function (msg) {
           backend.send(msg);
         });
 
-        backend.on('message',function (msg){
+        backend.on('message', function (msg) {
           frontend.send(msg);
         });
-
       }
       break;
     case 'router/dealer':
     case 'xrep/xreq':
-      if(capture){
-
-        //forwarding router/dealer pack signature: id, delimiter, msg
-        frontend.on('message',function (id,delimiter,msg){
-          backend.send([id,delimiter,msg]);
+      if (capture){
+        // forwarding router/dealer pack signature: id, delimiter, msg
+        frontend.on('message',function (id, delimiter, msg){
```

Comment:
also since we're in here doing style stuff can we do arrow functions for these callbacks?